pluginsValidate/isDarkmodeMaskCorrect.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+exports.type = 'validate';
+
+exports.name = 'isDarkmodeMaskCorrect';
+
+exports.active = true;
+
+exports.description =
+  'checks if the file has correct darkmodeMask color and size';
+
+const utils = require('./validationUtilities.js');
+
+/**
+ * Checks if the file has correct darkmodeMask color and size
+ *
+ * @example
+ *
+ * @param {Object} root current iteration root
+ * @param {Object} validateResult current validation result
+ * @param {Object} params current iteration params
+ *
+ * @return {Object} validateResult holding the validation result
+ *
+ * @author Jakub Filipowski
+ */
+
+exports.fn = function (root, validateResult, params) {
+  if (params || !utils.isEmpty(params)) {
+    const svg = utils.findElementByName(root, 'svg');
+    const darkmodeMaskElement = utils.findAllElementByAttributeValue(
+      svg,
+      'id',
+      'darkmodeMask'
+    )[0];
+
+    let isFillColorCorrect = false;
+
+    for (const color of params.fillColors) {
+      if (
+        utils.findAllElementByAttributeValue(darkmodeMaskElement, 'fill', color)
+          .length > 0 ||
+        darkmodeMaskElement.attributes.fill === color
+      ) {
+        isFillColorCorrect = true;
+      }
+    }
+
+    let isCorrectSize;
+    let elementWidth, elementHeight;
+    if (darkmodeMaskElement.name === 'path') {
+      [elementWidth, elementHeight] = utils.calculatePathDimensions(
+        darkmodeMaskElement.attributes.d
+      );
+    } else if (darkmodeMaskElement.name === 'g') {
+      // we assume that `g` elem has only one child that is path
+      const pathElem = darkmodeMaskElement.children.find(
+        (child) => child.name === 'path'
+      );
+
+      if (pathElem) {
+        [elementWidth, elementHeight] = utils.calculatePathDimensions(
+          pathElem.attributes.d
+        );
+      }
+    } else if (darkmodeMaskElement.name === 'ellipse') {
+      [elementWidth, elementHeight] = [
+        darkmodeMaskElement.attributes.rx * 2,
+        darkmodeMaskElement.attributes.ry * 2,
+      ];
+    } else if (darkmodeMaskElement.name === 'circle') {
+      [elementWidth, elementHeight] = [
+        darkmodeMaskElement.attributes.r * 2,
+        darkmodeMaskElement.attributes.r * 2,
+      ];
+    } else {
+      console.error(
+        'Error in plugin isDarkmodeMaskCorrect: Unhandled element found!'
+      );
+    }
+
+    if (elementWidth !== elementHeight) isCorrectSize = false;
+    else
+      isCorrectSize =
+        elementWidth === params.darkmodeFillSize[0] &&
```

Comment:
NOTE: poleciłbym gdzieś na górze desturkturyzacje tych params'ów aby to było czytelniejsze

Reply:
dobry pomysł, zrobione :)